File: codility_training/lesson.01.Iterations.BinaryGap/JunilHwang.java
import java.util.Arrays;

public class JunilHwang {
  public static int solution(int n) {
    String b = Integer.toBinaryString(n);
    int i = 0, max = 0, len = b.length();
    while (i < len) {
      int idx = b.indexOf("1", i + 1);
      if (idx == -1) break;
      max = Math.max(max, idx - i - 1);
      i = idx;
    }
    return max;
  }
}

File: codility_training/lesson.01.Iterations.BinaryGap/junilhwang.js
function solution (N) {
  const b = N.toString(2)
  const len = b.length
  let i = 0, max = 0;
  while (i < len) {
    const idx = b.indexOf('1', i + 1)
    if (idx === -1) break
    max = Math.max(idx - i - 1, max)
    i = idx
  }
  return max
}

console.log(solution(32) === 0)
console.log(solution(1041) === 5)
console.log(solution(9) === 2)
console.log(solution(529) === 4)
console.log(solution(20) === 1)
console.log(solution(15) === 0)